theano/gpuarray/sort.py

```diff
@@ -334,12 +360,18 @@ def local_gpua_topkop(op, ctx_name, inputs, outputs):
     ri = op.return_indices
     x, k = inputs
     x = as_gpuarray_variable(x, ctx_name)
-
+    if op.sorted:
+        return
     gpu_op = GpuTopKOp(
         axis=axis,
         sorted=op.sorted,
         idx_dtype=op.idx_dtype,
         return_values=rv,
         return_indices=ri)
-    rets = gpu_op(x, k)
+    rets = gpu_op(x, k, return_list=True)
+    c = ValuesEqApproxNoOrder(axis)
+    if rv or ri:
+        rets[0].tag.values_eq_approx = c
+    if rv and ri:
+        rets[1].tag.values_eq_approx = c
```

Comment:
PS: maybe a suggestion here:
```python
if rv:
    rets[0].tag.values_eq_approx = c
if ri:
    rets[1].tag.values_eq_approx = c
```

Reply:
Both are indenpendent. I'll loop over rets to make this more readable.